docs/spack.rst

```diff
@@ -0,0 +1,209 @@
+At shared compute facilities where Spack has been configured by the administrators,
+this may result in optimised builds without user intervention. With appropriate options,
+this also permits end users to customise binary optimisations by themselves.
+
+
+Prerequisites
+-------------
+
+This feature requires the `Spack <https://spack.io>`_ package manager to be installed on your system.
+
+
+How it works
+------------
+
+Nextflow  automatically creates and activates the Spack environment(s) given the dependencies
+specified by each process.
+
+Dependencies are specified by using the :ref:`process-spack` directive, providing either
+the names of the required Spack packages, the path of a Spack environment yaml file or
+the path of an existing Spack environment directory.
+
+.. note:: Spack always installs the software packages in its own directories, regardless of the Nextflow specifications.
+  The Spack environment created by Nextflow only contains symbolic links pointing to the appropriate package locations,
+  and therefore it is relatively small in size.
+
+You can specify the directory where the Spack environment is stored using the ``spack.cacheDir``
+configuration property (see the :ref:`configuration page <config-spack>` for details).
+When using a computing cluster, make sure to use a shared file system path
+accessible from all compute nodes.
+
+.. warning:: The Spack environment feature is not supported by executors that use
+  remote object storage as a work directory e.g. AWS Batch.
+
+Enabling Spack environment
+==========================
+
+The use of Spack recipes specified using the :ref:`process-spack`
+directive needs to be enabled explicitly by setting the option shown below in the pipeline
+configuration file (i.e. ``nextflow.config``)::
+
+    spack.enabled = true
+
+
+Alternatively it can be specified by setting the variable ``NXF_SPACK_ENABLED=true`` in your environment
+or by using the ``-with-spack true`` command line option.
+
+
+Use Spack package names
+=======================
+
+Spack package names can specified using the ``spack`` directive. Multiple package names can be specified
+by separating them with a blank space.
+For example::
+
+  process foo {
+    spack 'bwa samtools py-multiqc'
+
+    '''
+    your_command --here
+    '''
+  }
+
+Using the above definition a Spack environment that includes BWA, Samtools and MultiQC tools is created and
+activated when the process is executed.
+
+The usual Spack package syntax and naming conventions can be used. The version of a package can be
+specified after the package name as shown here ``bwa@0.7.15``.
+
+Optimisation for the local CPU microarchitetcure can be requested by adding the option ``target=<LOCAL ARCH>``
+after a package name. For instance, if the compute infrastructure uses AMD Zen3 microprocessors,
+use the following to optimise all packages for it: ``bwa target=zen3 samtools target=zen3 py-multiqc target=zen3``.
```

Comment:
Is the `target=something` separate by a blank a Spack convention? it looks a bit confusing because also packages are separated by a blank

Reply:
Yes that is how it is done in Spack.
There is actually more to this.
In general, you define a package installation request by means of a package(s) "specification", or `spec`.
A `spec` can be as verbose as (I am making up the string values):
```
lammps@22.01 %gcc@12 ^fftw@3 plumed=on gpu=cuda cxxflags=-O3 target=zen3
```
And then you can pair multiple packages `spec` in a single request.

Clearly, a better option for clarity is to use a YAML rather than a one-liner (spoiler alert, it can be MAAANY lines!!) for a package(s) request.

Reply:
fine!